investment-app/TRINV.Api/TRINV.Domain/Validations/Ensure.cs

```diff
@@ -0,0 +1,74 @@
+﻿namespace TRINV.Domain.Validations;
+
+using System.Runtime.CompilerServices;
+using System.Text.RegularExpressions;
+
+public static class Ensure
+{
+    public static void IsArgumentNull(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (value == null)
+        {
+            throw new ArgumentException("The argument value can't be null", paramName);
+        }
+    }
+
+    public static void IsArgumentNullOrEmpty(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (string.IsNullOrEmpty(value))
+        {
+            throw new ArgumentException("The argument value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsArgumentNullOrWhiteSpace(string? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (string.IsNullOrWhiteSpace(value))
+        {
+            throw new ArgumentException("The argument value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsNull(object? value,
+        [CallerArgumentExpression("value")] string? paramName = null)
+    {
+        if (value is null)
+        {
+            throw new ArgumentException("Object value can't be null or empty", paramName);
+        }
+    }
+
+    public static void IsEqual(
```

Comment:
As we discussed. Did you checkl how IsEqual is working with object and primitive types

Reply:
Done.